decidim-core/lib/decidim/core/test/factories.rb

```diff
@@ -162,6 +166,35 @@
                role: :admin
       end
     end
+
+    trait :process_collaborator do
+      transient { participatory_process nil }
+
+      after(:create) do |user, evaluator|
+        create :participatory_process_user_role,
+               user: user,
+               participatory_process: evaluator.participatory_process,
+               role: :collaborator
+      end
+    end
+
+    trait :process_moderator do
+      transient { participatory_process nil }
+
+      after(:create) do |user, evaluator|
+        create :participatory_process_user_role,
+               user: user,
+               participatory_process: evaluator.participatory_process,
+               role: :moderator
+      end
+    end
+
```

Comment:
Yes, the investigation I made about coverage yesterday found exactly that! :)

* The moderator role is completely uncovered, everywhere.
* "widget controllers" are completely uncovered too. As a matter of fact, since they are autoloaded, they are not even required during the whole test run.

Maybe we can open some issues about this.